core/array/slice_spec.rb

```diff
@@ -120,6 +120,27 @@ def to.to_int() -2 end
     -> { a.slice!(from .. "b") }.should raise_error(TypeError)
   end
 
+  it "raises TypeError if to_int returns non-integer" do
+    from = mock('from')
+    to = mock('to')
+
+    # So we can construct a range out of them...
+    def from.<=>(o) 0 end
+    def to.<=>(o) 0 end
+
+    a = [1, 2, 3, 4, 5]
+
+    def from.to_int() 'cat' end
+    def to.to_int() -2 end
+
+    -> { a.slice(from..to) }.should raise_error(TypeError)
+
+    def from.to_int() 1 end
+    def to.to_int() 'cat' end
+
+    -> { a.slice(from..to) }.should raise_error(TypeError)
+  end
```

Comment:
Looks good, but this should be in `core/array/shared/slice.rb` to test both for `slice` and for `[]`

Reply:
@bolshakov Could you update the PR, moving this to core/array/shared/slice.rb?

Reply:
@eregon sure. Moved 👌